packages/esm-patient-test-results-app/src/overview/common-overview.component.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback } from 'react';
+import Table16 from '@carbon/icons-react/es/table/16';
+import ChartLine16 from '@carbon/icons-react/es/chart--line/16';
+import { Button, TableToolbarContent, TableToolbar } from 'carbon-components-react';
+import { EmptyState } from '@openmrs/esm-patient-common-lib';
+import { headers, formatDate, InfoButton, Separator } from './helpers';
+import { OverviewPanelEntry } from './useOverviewData';
+import { useTranslation } from 'react-i18next';
+import { navigate } from '@openmrs/esm-framework';
+import CommonDataTable from './common-datatable.component';
+import styles from './common-overview.scss';
+
+const DashboardResultsCount = 5;
+
+interface CommonOverviewPropsBase {
+  overviewData: Array<OverviewPanelEntry>;
+  insertSeparator?: boolean;
+  patientUuid?: string;
+}
+
+interface CommonOverviewPropsWithToolbar {
+  openTimeline: (panelUuid: string) => void;
+  openTrendline: (panelUuid: string, testUuid: string) => void;
+}
+
+interface CommonOverviewPropsWithoutToolbar {
+  hideToolbar: true;
+}
+
+type Only<T, U> = {
+  [P in keyof T]: T[P];
+} & {
+  [P in keyof U]?: never;
+};
+
+type Either<T, U> = Only<T, U> | Only<U, T>;
+
+type CommonOverviewProps = CommonOverviewPropsBase &
+  Either<CommonOverviewPropsWithToolbar, CommonOverviewPropsWithoutToolbar>;
+
+const CommonOverview: React.FC<CommonOverviewProps> = ({
+  overviewData = [],
+  openTimeline,
+  openTrendline,
+  insertSeparator = false,
+  hideToolbar = false,
+  patientUuid,
+}) => {
+  const { t } = useTranslation();
+  const [activeCardUuid, setActiveCardUuid] = React.useState('');
+
+  const isActiveCard = useCallback(
+    (uuid: string) =>
+      activeCardUuid === uuid || (!activeCardUuid && uuid === overviewData[0][overviewData[0].length - 1]),
+    [activeCardUuid, overviewData],
+  );
+
+  const handleSeeAvailableResults = useCallback(() => {
+    navigate({ to: `\${openmrsSpaBase}/patient/${patientUuid}/chart/test-results` });
+  }, [patientUuid]);
+
+  if (!overviewData.length)
+    return <EmptyState headerTitle={t('testResults', 'Test Results')} displayText={t('testResults', 'test results')} />;
+
+  return (
+    <>
+      {(() => {
+        const cards = overviewData.map(([title, type, data, date, uuid]) => (
+          <article
```

Comment:
Nice, now that I'm reviewing the Mozilla docs for `article`, I'm realizing that maybe we should be using that more often in extensions.